pywbem_mock/_providerdispatcher.py

```diff
@@ -95,207 +97,580 @@ def default_method_provider(self):
-                    PropertyList=PropertyList)
-            except AttributeError:
-                pass
-
-        return self.default_instance_write_provider.ModifyInstance(
-            ModifiedInstance,
-            IncludeQualifiers=IncludeQualifiers,
-            PropertyList=PropertyList)
+        # Call the provider method.
+        result = provider.ModifyInstance(
+            modified_instance, IncludeQualifiers=IncludeQualifiers)
+
+        # Verify provider method result.
+        assert result is None
 
     def DeleteInstance(self, InstanceName):
         # pylint: disable=invalid-name
         """
-        Dispatcher for the DeleteInstance method.
+        Dispatcher for the DeleteInstance provider method.
 
-        This method validates input parameters against the repository and
-        provider required types and if successful, routes the method either to
-        the default provider or to a provider registered for the namespace and
-        class in InstanceName
+        This method performs validations and if successful, routes the provider
+        method call either to a registered provider, or to the default provider.
         """
 
-        # Validate input parameters
+        # Verify the input parameter types (type errors have already been
+        # raised during checks in the WBEMConnection operation).
+        assert isinstance(InstanceName, CIMInstanceName)
+
+        # Verify that the namespace exists in the CIM repository.
         namespace = InstanceName.namespace
         self.validate_namespace(namespace)
 
-        # Test if corresponding class and instance already exist
-        if not self.class_exists(namespace, InstanceName.classname):
+        class_store = self.cimrepository.get_class_store(namespace)
+        instance_store = self.cimrepository.get_instance_store(namespace)
+
+        # Verify that the creation class of the instance to be deleted exists
+        # in the CIM repository.
+        if not class_store.object_exists(InstanceName.classname):
             raise CIMError(
                 CIM_ERR_INVALID_CLASS,
-                _format("Class {0!A} in namespace {1!A} not found. "
-                        "Cannot delete instance {2!A}",
-                        InstanceName.classname, namespace, InstanceName))
+                _format("Creation class {0!A} of instance to be deleted does "
+                        "not exist in namespace {1!A} of the CIM repository.",
+                        InstanceName.classname, namespace))
 
-        instance_store = self.cimrepository.get_instance_store(namespace)
+        # Verify that the instance to be deleted exists in the CIM repository.
         if not instance_store.object_exists(InstanceName):
             raise CIMError(
                 CIM_ERR_NOT_FOUND,
-                _format("Instance {0!A} not found in CIM repository",
+                _format("Instance to be deleted does not exist in the CIM "
```

Comment:
Generally we place data within the text stream.  In this case the path is at the end instead of in the text.   Just checking since one logical reason for that is that paths are long and messy.

Reply:
ANSWER
I did it that way because the path is long and messy, and this particular message text allowed to do that easily.